### PR TITLE
[v9.2.x] Chore: Upgrade Go to 1.20.1 and Alpine to 3.17 (#63891)

### PR DESCRIPTION
**Pull request by dsotirakis**
(cherry picked from commit 4e687dfb6f050cac7e6b1622e5050992dad13829) (cherry picked from commit 4f0993b346a8dc0281400ed28f81a3c3206992f6)

Backports: #63891

### CODE REVIEW

File: e2e/cloud-plugins-suite/azure-monitor.spec.ts
Comment (sakjur) [inline e2e/cloud-plugins-suite/azure-monitor.spec.ts:1]: Where does this file come from? Should it be here?

**Review by sakjur — APPROVED** [review]
Seems to work for me locally when https://github.com/grafana/grafana/pull/66445/files#r1165468049 is fixed :+1:

File: e2e/cloud-plugins-suite/azure-monitor.spec.ts
Comment (dsotirakis) [reply e2e/cloud-plugins-suite/azure-monitor.spec.ts:1]: Yeah, backporting CI changes means that we need to backport the azure e2e tests additions. They are not needed/not working for 9.2 but if we want to be consistent they need to be in the drone yaml.
I am skipping all of the tests though so it should be fine.
(cc @aangelisc)

Comment (sakjur) [reply e2e/cloud-plugins-suite/azure-monitor.spec.ts:1]: Ah, that makes sense :+1:

**Pull request merged**